Add link to request for library update and report outdated library

sufuf3 opened the pull request:
cc #145
Add a request update link on library page

![image](https://cloud.githubusercontent.com/assets/8349587/23585311/5f2a454a-01b6-11e7-9d97-971dea4834cc.png)

click link and redirect to GitHub page

![image](https://cloud.githubusercontent.com/assets/8349587/23585425/2aeb4942-01ba-11e7-9448-87e520f57750.png)

PeterDaveHello commented on templates/library.html:
```diff
@@ -38,7 +38,7 @@ <h1 style="margin-top: 00px;" id="libraryName">{{library.name}}</h1>
         </div>
       </div>
-       <div class="input-group version-dropdown">
+        <div class="input-group version-dropdown">
```
Unrelated diff ...

PeterDaveHello commented:
Should give a screenshot and example link.

PeterDaveHello commented:
Wrong syntax in commit message.

sufuf3 commented:
example link: /libraries/jquery

PeterDaveHello commented:
1) You didn't provide the link
2) You should provide it in the first comment
3) The design of the place and style of the link ... makes itself looks like a broken, I don't think we should make it appear in that place and in that way.

sufuf3 commented:
@cdnjs/intern3 Please help me review this PR, thanks.

I change this request update link place on library page
![image](https://cloud.githubusercontent.com/assets/8349587/23586651/76022f62-01d4-11e7-8eb6-3a44750fe205.png)


And redirect to GitHub page as following:
![image](https://cloud.githubusercontent.com/assets/8349587/23586661/93cadc9c-01d4-11e7-9042-88d7ae5c63c4.png)

If there is any suggest, please let me know.

PeterDaveHello commented:
It's a bad idea to simply clone the template from new library request, totally different purpose.

sufuf3 commented:
@PeterDaveHello I pulled from this repo. , please help me review this PR.
And if there is any suggest, please let me know.
Thank you.

PeterDaveHello commented:
I mean the issue template, it's originally for new library request, you should not just copy and paste as the purpose is different.

sufuf3 commented:
What items should issue template take is better?
How about as following? (`XXXXX` will auto add from library API.)

**Library name:** XXXXX
**CDNJS version:** XXXXX
**CDNJS URL:** XXXXX
**Git repository url:** XXXXX
**npm package name or url (if there is one):** XXXXX
**Need update items and description:**
 * [ ] **description:** XXXXX -> _____________________
 * [ ] **version:**   XXXXX -> _____________________
 * [ ] **author:**    XXXXX -> _____________________
 * [ ] **filename:**     XXXXX -> _____________________
 * [ ] **homepage:**     XXXXX -> _____________________
 * [ ] **repository:**     XXXXX -> _____________________
 * [ ] **keywords:**     XXXXX -> _____________________
 * [ ] **license:**     XXXXX -> _____________________

**Wanna say something? Leave message here:** 

If there is any suggest, please let me know.

PeterDaveHello commented:
@sufuf3 It's much more better that you started to think about its content now, and you should also explain the reason why or why not the items would be or would not be listed here.

sufuf3 commented:
I refer to `package.json`.
In **Need update items and description:**, I include all of the items, exclude as following:
1. `repository`, because it show in **Git repository URL:** 
2. `name`, it show in **Library name:**
3. `auto-update`, this item is hard to detect, cause it may not exist, from npm to git repo., or from git repo, to npm. I think it can let the user leave a message.

If there is any suggest, please let me know.
Thank you.

sufuf3 commented:
Or how about as following? (`XXXXX` will auto add from library API.)

**Library name:** XXXXX
**CDNJS version:** XXXXX
**CDNJS URL:** XXXXX
**Git repository url:** XXXXX
**npm package name or url (if there is one):** XXXXX
**Need update items and description:**
 * [ ] **description:** XXXXX -> _____________________
 * [ ] **version:**   XXXXX -> _____________________
 * [ ] **author:**    XXXXX -> _____________________
 * [ ] **filename:**     XXXXX -> _____________________
 * [ ] **homepage:**     XXXXX -> _____________________
 * [ ] **repository:**     XXXXX -> _____________________
 * [ ] **keywords:**     XXXXX -> _____________________
 * [ ] **license:**     XXXXX -> _____________________
 * [ ] **auto-update:** 
   * [ ] None to npm auto-update
   * [ ] None to git auto-update
   * [ ] git auto-update to npm auto-update
   * [ ] npm auto-update to git auto-update

**Wanna say something? Leave message here:**

sufuf3 commented:
@hare1039 Could you give me some suggestion and make this PR to be better?

clairetsai818 commented:
**- [ ] This library is not updated to the latest version

- [ ] The GitHub repo has changed

- [ ] Some info of this lib needs to be updated
       Info needed to be updated:**

I think that we can list out the reasons that happens often, so that the issuer only needs to check that box.

sufuf3 commented:
@clairetsai818 Thank you for your suggestion. I modified as the following, hope it is okay.

**Library name:** XXXXX
**CDNJS version:** XXXXX
**CDNJS URL:** XXXXX
**Git repository url:** XXXXX
**npm package name or url (if there is one):** XXXXX

**Update Reason:**
 * [ ] **This library is not updated to the latest version.**
 * [ ] **The GitHub repo has changed.**
 * [ ] **Some info of this lib needs to be updated:**

   * [ ] **description:** XXXXX -> _____________________
   * [ ] **version:**   XXXXX -> _____________________
   * [ ] **author:**    XXXXX -> _____________________
   * [ ] **filename:**     XXXXX -> _____________________
   * [ ] **homepage:**     XXXXX -> _____________________
   * [ ] **repository:**     XXXXX -> _____________________
   * [ ] **keywords:**     XXXXX -> _____________________
   * [ ] **license:**     XXXXX -> _____________________

**Wanna say something? Leave message here:**

hare1039 commented:
> This library is not updated to the latest version.

A little bit too long. How about `This library needs update to latest version`
> The GitHub repo has changed.

Not necessary GitHub I think.

Will those `XXXXX` auto filled?

sufuf3 commented:
I modified it.
And I think the redirect url to the GitHub page template maybe can using as the following: 

**Library name:** XXXXX
**CDNJS version:** XXXXX
**CDNJS URL:** XXXXX
**Git repository url:** XXXXX
**npm package name or url (if there is one):** XXXXX

**Update Reason:**
   * [ ] **description:** XXXXX -> _____________________
   * [ ] **version:**   XXXXX -> _____________________
   * [ ] **author:**    XXXXX -> _____________________
   * [ ] **filename:**     XXXXX -> _____________________
   * [ ] **homepage:**     XXXXX -> _____________________
   * [ ] **repository:**     XXXXX -> _____________________
   * [ ] **keywords:**     XXXXX -> _____________________
   * [ ] **license:**     XXXXX -> _____________________

**Wanna say something? Leave message here:** 

----

I refer to `package.json`.
In **Need update items and description:**, I include all of the items, exclude as following:
1. `repository`, because it show in **Git repository URL:** 
2. `name`, it show in **Library name:**
3. `auto-update`, this item is hard to detect, cause it may not exist, from npm to git repo., from git repo. to npm or file structure problem. I think it can let the user leave a message. And we get **Git repository url** and **npm package name or url** info, it let us know faster.
3.  XXXXX will auto add from library API.

If there is any suggest, please let me know.
Thank you.

gocreating commented:
It will be better readability with indention

sufuf3 commented:
@gocreating thanks, I'll try to modify this.
sorry for late response.

qas612820704 commented:
Hi @sufuf3, it seems like #71 can be done by this PR at the same time.


By modifying to @gocreating 's code like
```js
{{
  let getURL = (type, ...args) => {
    if (type == 'update')
      return `https://github.com/cdnjs/cdnjs/issues/new?title=${...}&body=${...}&...`;
    else if (type == 'report')
      return ...;
   };
}}
<a href="{{ getURL('update', ...) }}">
  request update
</a>
<a href="{{ getURL('report', ...) }}">
  report lib
</a>
```

How about this? :)

sufuf3 commented:
@qas612820704 Thank you very much.
I'll try to use it.

sufuf3 commented:
@PeterDaveHello we added two links as the followings:
1. request for library update #145 
2. report outdated library #71 

- This is the page show the links:
![image](https://cloud.githubusercontent.com/assets/8349587/26527535/95cc8d48-43c8-11e7-883a-ead87335627c.png)
- This is the link redirect to GitHub to request for library update:
![image](https://cloud.githubusercontent.com/assets/8349587/26527574/1678f526-43c9-11e7-84e8-fa76f02c2dca.png)
- This is the link redirect to GitHub to report outdated library:
![image](https://cloud.githubusercontent.com/assets/8349587/26527600/c081082e-43c9-11e7-91fc-b3995bd136fc.png)


If you have any suggestion, please let me know.
Thank you.

PeterDaveHello commented:
We don't need so many items, just its version.
Also I don't think we need user to fill repo or hosted version info.

sufuf3 commented:
I modified as the followings.
If you have any suggestion, please let me know.
Thanks.

* This is the page show the links:
![image](https://cloud.githubusercontent.com/assets/8349587/26565446/ec5c4efe-451d-11e7-89a9-8ce4d909bbce.png)
* This is the link redirect to GitHub to request for library update:
![image](https://cloud.githubusercontent.com/assets/8349587/26565511/6dc5450e-451e-11e7-8b9b-c0019f416ce3.png)
* This is the link redirect to GitHub to report outdated library:
![image](https://cloud.githubusercontent.com/assets/8349587/26565520/801f891c-451e-11e7-93ca-5d413b135da3.png)

PeterDaveHello commented:
It's confused about what's the difference from the name of the title/button, also, I don't like the place where they are placed, the button(s) should be individual, not mixed with the dropdown list.

PeterDaveHello commented:
This situation is not good to use `wanna say something`, it's about the description of what the request is.

sufuf3 commented:
I modified as the followings.
If you have any suggestion, please let me know.
Thanks.

- modified the button part
![image](https://user-images.githubusercontent.com/8349587/27382894-4e6e91a2-56bb-11e7-8d99-56f13e80f5c0.png)

-  remove `wanna say something` part of the description.
![image](https://user-images.githubusercontent.com/8349587/27382976-89c51050-56bb-11e7-81d2-2d325dd06da9.png)
